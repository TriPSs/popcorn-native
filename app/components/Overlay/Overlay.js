import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, View } from 'react-native'

export const styles = StyleSheet.create({

  root: {
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    backgroundColor: 'black',
    opacity: 0.50,
  },

  default: {},

  dark: {
    opacity: 0.70,
  },

})

export const Overlay = ({ style, variant }) => (
  <View style={[styles.root, styles[variant], style]} />
)

Overlay.propTypes = {
  variant: PropTypes.oneOf([
    'default',
    'dark',
  ]),
}

Overlay.defaultProps = {
  variant: 'default',
  style: {},
}

export default Overlay
