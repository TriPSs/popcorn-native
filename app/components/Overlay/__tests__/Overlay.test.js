import React from 'react'
import { shallow } from 'enzyme'

import Overlay from '../Overlay'

describe('(Component) Overlay', () => {

  it('should render with defaults', () => {
    const result = shallow(
      <Overlay />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render dark variant', () => {
    const result = shallow(
      <Overlay variant={'dark'} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with animation', () => {
    const result = shallow(
      <Overlay withAnimation />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with custom style', () => {
    const result = shallow(
      <Overlay style={{ margin: 5 }} />,
    )

    expect(result).toMatchSnapshot()
  })

})
