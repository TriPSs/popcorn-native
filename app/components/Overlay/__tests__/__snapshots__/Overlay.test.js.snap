// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Overlay should render dark variant 1`] = `
<withAnimatable(View)
  animation={null}
  delay={0}
  direction="normal"
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  useNativeDriver={true}
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "black",
          "opacity": 0.7,
        },
        Object {},
      ]
    }
  />
</withAnimatable(View)>
`;

exports[`(Component) Overlay should render with animation 1`] = `
<withAnimatable(View)
  animation="fadeIn"
  delay={0}
  direction="normal"
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  useNativeDriver={true}
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "black",
          "opacity": 0.5,
        },
        Object {},
      ]
    }
  />
</withAnimatable(View)>
`;

exports[`(Component) Overlay should render with custom style 1`] = `
<withAnimatable(View)
  animation={null}
  delay={0}
  direction="normal"
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  useNativeDriver={true}
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "black",
          "opacity": 0.5,
        },
        Object {
          "margin": 5,
        },
      ]
    }
  />
</withAnimatable(View)>
`;

exports[`(Component) Overlay should render with defaults 1`] = `
<withAnimatable(View)
  animation={null}
  delay={0}
  direction="normal"
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  useNativeDriver={true}
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "black",
          "opacity": 0.5,
        },
        Object {},
      ]
    }
  />
</withAnimatable(View)>
`;
